scrunch/datasets.py

```diff
@@ -141,23 +141,30 @@ def _get_dataset(dataset, connection=None, editor=False, project=None):
         else:
             shoji_ds = project.get_dataset(dataset).resource
     else:
-        # search by dataset name
-        try:
-            shoji_ds = root_datasets.by('name')[dataset].entity
-        except KeyError:
-            # search by dataset id
+        # Check if dataset reference is a valid uuid, if so, try to load it
+        # directly
+        if validate_uuid(dataset):
```

Comment:
Do not rely on the ID to be a UUID, it may not be in the future (if we move towards shorter IDs). I suggest you just always try to get by ID first, and if it doesn't work then do the other strategy.